src/components/WeeklyCalendar/index.d.ts

```diff
@@ -0,0 +1,24 @@
+import { BaseProps } from '../types';
+
+export interface CalendarEvent {
+    title: string;
+    description?: string;
+    startDate: Date;
+    endDate: Date;
+}
+
+export interface WeeklyCalendarEvent {
+    week: Date;
+}
+
+export interface WeeklyCalendarProps extends BaseProps {
+    events: Array<CalendarEvent>;
+    currentWeek?: Date | string;
+    minDate?: Date;
+    maxDate?: Date;
+    onWeekChange: (event: WeeklyCalendarEvent) => void;
+    onEventClick: (event: CalendarEvent) => void;
```

Comment:
I think all the props are optional, components should avoid having required props, because what required really means?? the components should always have fallback values